# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmGenerator.cpp
# Commit: 5718a8d121f9
# Full Hash: 5718a8d121f95ce740a11ee443444cea990d6323
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 7: Remove Frame::tls. r=lth
#   
#   We remove Frame::tls and fix prologue/epilogue. Runtime, Ion and Baseline are ready for this move already.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83064
# ==============================================================================

diff -r 40031ad88dbd -r 5718a8d121f9 js/src/wasm/WasmGenerator.cpp
--- a/js/src/wasm/WasmGenerator.cpp	Tue Nov 03 10:28:10 2020 +0000
+++ b/js/src/wasm/WasmGenerator.cpp	Tue Nov 03 10:32:55 2020 +0000
@@ -558,11 +558,8 @@
       case CallSiteDesc::LeaveFrame: {
         Uint32Vector& jumps = metadataTier_->debugTrapFarJumpOffsets;
         if (jumps.empty() || !InRange(jumps.back(), callerOffset)) {
-          // See BaseCompiler::insertBreakablePoint for why we must
-          // reload the TLS register on this path.
           Offsets offsets;
           offsets.begin = masm_.currentOffset();
-          masm_.loadPtr(Address(FramePointer, Frame::tlsOffset()), WasmTlsReg);
           CodeOffset jumpOffset = masm_.farJumpWithPatch();
           offsets.end = masm_.currentOffset();
           if (masm_.oom()) {